test/json_formatter_SUITE.erl

```diff
@@ -0,0 +1,392 @@
+                text => "JSON-match-this-struct-depth-limited",
+                deep_struct => DeepStruct}),
+
+    After = case get_at_least_n_log_lines(?LOGFILE,
+                                          length(Before) + 1,
+                                          timer:seconds(1)) of
+                timeout -> ct:fail("timed out waiting for messages to reach the log file");
+                Res -> Res
+            end,
+
+    [Line] = filter_out_non_matching(After -- Before, <<"JSON-match-this-struct-depth-limited">>),
+
+    Decoded = jiffy:decode(Line, [return_maps]),
+
+    ShortenedStruct = unicode:characters_to_binary(io_lib:format("~0P", [DeepStruct, FormatDepth])),
+
+    #{<<"level">> := <<"info">>,
+      <<"when">> := _Time,
+      <<"meta">> := #{<<"file">> := _File,
+                      <<"line">> := _Line,
+                      <<"mfa">> := _MFA,
+                      <<"pid">> := _Pid},
+      <<"what">> := <<"format_depth_limited">>,
+      <<"text">> := <<"JSON-match-this-struct-depth-limited">>,
+      <<"deep_struct">> := ShortenedStruct}
+    = Decoded.
+
+
+json_depth_limited(Config) ->
+
+    {_, File} = ?config(logger_backend, Config),
+    Before = get_log(File),
+
+    Depth = ?config(depth, Config),
+    DeepList = deep_list(Depth*2),
+
+    ?LOG_INFO(#{what => json_depth_limited,
+                text => "JSON-match-this-json-depth-limited",
+                deep_list => DeepList}),
+
+    After = case get_at_least_n_log_lines(?LOGFILE,
+                                          length(Before) + 1,
+                                          timer:seconds(1)) of
+                timeout -> ct:fail("timed out waiting for messages to reach the log file");
+                Res -> Res
+            end,
+
+    [Line] = filter_out_non_matching(After -- Before, <<"JSON-match-this-json-depth-limited">>),
+
+    Decoded = jiffy:decode(Line, [return_maps]),
+
+    % 1 because it is at the first level
+    ShortenedList = deep_list(Depth - 1, <<"...">>),
+
+    #{<<"level">> := <<"info">>,
+      <<"when">> := _Time,
+      <<"meta">> := #{<<"file">> := _File,
+                      <<"line">> := _Line,
+                      <<"mfa">> := _MFA,
+                      <<"pid">> := _Pid},
+      <<"what">> := <<"json_depth_limited">>,
+      <<"text">> := <<"JSON-match-this-json-depth-limited">>,
+      <<"deep_list">> := ShortenedList}
+    = Decoded.
+
+%%
+%% Helpers
+%%
+
+example_acc(Body) ->
+    #{lserver => <<"localhost">>,
+      mongoose_acc => true,
+      non_strippable => [],
+      origin_location => #{file => "/Users/user/MongooseIM/src/ejabberd_router.erl",
+                           line => 116,
+                           mfa => {ejabberd_router,route,3}},
+      origin_pid => self(),
+      origin_stanza => <<<<"<message type='chat' id='1111'><body>">>/binary,
+                         Body/binary,
+                         <<"</body></message>">>/binary>>,
```

Comment:
Why not
```suggestion
      origin_stanza => <<"<message type='chat' id='1111'><body>",
                         Body/binary,
                         "</body></message>">>,
```

Reply:
Thanks, I hate the binary syntax and its documentation. I will change it in all places.